Tests/FluentAssertions.Specs/Specialized/TaskOfTAssertionSpecs.cs

```diff
@@ -165,6 +164,28 @@ public async Task When_task_completes_late_it_in_assertion_scope_should_fail()
             await action.Should().ThrowAsync<XunitException>();
         }
 
+        [Fact]
+        public async Task When_task_does_not_completes_the_result_extension_does_not_hang()
+        {
+            // Arrange
+            var timer = new FakeClock();
+            var taskFactory = new TaskCompletionSource<int>();
+
+            // Act
+            Func<Task> action = () =>
+            {
+                Func<Task<int>> func = () => taskFactory.Task;
+                using var _ = new AssertionScope();
+                return func.Should(timer).CompleteWithinAsync(100.Milliseconds()).WithResult(2);
+            };
+            timer.Complete();
+
+            // Assert
+            var assertionTask = action.Should().ThrowAsync<XunitException>()
+                .WithMessage("Expected*to complete within 100ms.*Expected return*to be 2, but found 0.");
+            await this.Awaiting(x => assertionTask).Should().CompleteWithinAsync(200.Seconds());
```

Comment:
Adding `using static FluentAssertions.FluentActions;`

```suggestion
            await Awaiting(() => assertionTask).Should().CompleteWithinAsync(200.Seconds());
```